compiler-core/src/type_.rs

```diff
@@ -641,6 +643,11 @@ impl ModuleInterface {
     pub fn contains_todo(&self) -> bool {
         self.warnings.iter().any(|warning| warning.is_todo())
     }
+
+    pub fn remove_duplicated_type_aliases(&mut self) {
+        self.types
+            .retain(|name, _| !self.type_aliases.contains_key(name));
+    }
```

Comment:
What is this function for? Seems to be called a lot.

Reply:
As during the type analysis aliases are registered as normal types, it needs to be removed later. I think I tried removing it only after analysis but there were still type aliases registered as types. I guess there's a better way to do this, although I haven't found how.

Reply:
They should not be removed, no. They are types in the module.